code/modules/wiremod/components/admin/filter.dm

```diff
@@ -0,0 +1,313 @@
+	),
+	"rays" = list(
+		"x" = PORT_TYPE_NUMBER,
+		"y" = PORT_TYPE_NUMBER,
+		"size" = PORT_TYPE_NUMBER,
+		"color" = PORT_TYPE_STRING,
+		"offset" = PORT_TYPE_NUMBER,
+		"density" = PORT_TYPE_NUMBER,
+		"threshold" = PORT_TYPE_NUMBER,
+		"factor" = PORT_TYPE_NUMBER,
+		"flags" = PORT_TYPE_NUMBER,
+	),
+	"ripple" = list(
+		"x" = PORT_TYPE_NUMBER,
+		"y" = PORT_TYPE_NUMBER,
+		"size" = PORT_TYPE_NUMBER,
+		"repeat" = PORT_TYPE_NUMBER,
+		"radius" = PORT_TYPE_NUMBER,
+		"falloff" = PORT_TYPE_NUMBER,
+		"flags" = PORT_TYPE_NUMBER,
+	),
+	"wave" = list(
+		"x" = PORT_TYPE_NUMBER,
+		"y" = PORT_TYPE_NUMBER,
+		"size" = PORT_TYPE_NUMBER,
+		"offset" = PORT_TYPE_NUMBER,
+		"flags" = PORT_TYPE_NUMBER,
+	),
+))
+
+GLOBAL_LIST_INIT(wiremod_flag_info, list(
+	"Animation Easing" = list(
+		"Easing Type" = list(
+			"LINEAR_EASING" = LINEAR_EASING,
+			"SINE_EASING" = SINE_EASING,
+			"CIRCULAR_EASING" = CIRCULAR_EASING,
+			"QUAD_EASING" = QUAD_EASING,
+			"CUBIC_EASING" = CUBIC_EASING,
+			"BOUNCE_EASING" = BOUNCE_EASING,
+			"ELASTIC_EASING" = ELASTIC_EASING,
+			"BACK_EASING" = BACK_EASING,
+			"JUMP_EASING" = JUMP_EASING,
+		),
+		"EASE_IN" = EASE_IN,
+		"EASE_OUT" = EASE_OUT,
+	),
+	"Animation Flags" = list(
+		"ANIMATION_END_NOW" = ANIMATION_END_NOW,
+		"ANIMATION_LINEAR_TRANSFORM" = ANIMATION_LINEAR_TRANSFORM,
+		"ANIMATION_PARALLEL" = ANIMATION_PARALLEL,
+		"ANIMATION_RELATIVE" = ANIMATION_RELATIVE,
+	),
+	"Alpha Filter Flags" = list(
+		"MASK_INVERSE" = MASK_INVERSE,
+		"MASK_SWAP" = MASK_SWAP,
+	),
+	"Color Space" = list("_" = list(
+		"COLORSPACE_RGB" = COLORSPACE_RGB,
+		"COLORSPACE_HSV" = COLORSPACE_HSV,
+		"COLORSPACE_HSL" = COLORSPACE_HSL,
+		"COLORSPACE_HCY" = COLORSPACE_HCY,
+	)),
+	"Layering Flags" = list("_" = list(
+		"FILTER_OVERLAY" = FILTER_OVERLAY,
+		"FILTER_UNDERLAY" = FILTER_UNDERLAY,
+	)),
+	"Layering Blend Mode" = list("_" = list(
+		"BLEND_OVERLAY" = BLEND_OVERLAY,
+		"BLEND_ADD" = BLEND_ADD,
+		"BLEND_SUBTRACT" = BLEND_SUBTRACT,
+		"BLEND_MULTIPLY" = BLEND_MULTIPLY,
+		"BLEND_INSET_OVERLAY" = BLEND_INSET_OVERLAY,
+	)),
+	"Outline Flags" = list(
+		"OUTLINE_SHARP" = OUTLINE_SHARP,
+		"OUTLINE_SQUARE" = OUTLINE_SQUARE,
+	),
+	"Ray Flags" = list(
+		"FILTER_OVERLAY" = FILTER_OVERLAY,
+		"FILTER_UNDERLAY" = FILTER_UNDERLAY,
+	),
+	"Ripple Flags" = list(
+		"WAVE_BOUNDED" = WAVE_BOUNDED,
+	),
+	"Wave Flags" = list(
+		"WAVE_BOUNDED" = WAVE_BOUNDED,
+		"WAVE_SIDEWAYS" = WAVE_SIDEWAYS,
+	),
+))
+
+/obj/item/circuit_component/filter_helper
+	display_name = "Filter Parameter Helper"
+	desc = "Constructs a list of filter parameters from the inputs."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL|CIRCUIT_FLAG_ADMIN
+
+	var/datum/port/input/option/filter_type_port
+	var/current_filter_type
+
+	var/list/filter_params = list()
+
+	var/datum/port/output/output_params
+
+/obj/item/circuit_component/filter_helper/populate_options()
+	filter_type_port = add_option_port("Filter Type", GLOB.wiremod_filter_info)
```

Comment:
could you only have a list of the keys here so that it isn't sending all that data to the client?

Reply:
you did it for the other one but not this one

Reply:
see here

Reply:
```suggestion
	filter_type_port = add_option_port("Filter Type", assoc_to_keys(GLOB.wiremod_filter_info))
```